### PR TITLE
Support editing composer to be used as main composer (feature flagged)

### PR DESCRIPTION
**Pull request by bwindels**
https://github.com/vector-im/riot-web/issues/9866

Uses the new editor (CIDER) developed for editing messages to power the main composer.
It is feature flagged behind `feature_cider_composer`, which will need to be put to `"labs"` on /develop.

The new composer does not support rich text editing, only markdown, but should otherwise be mostly on par with the slate editor. See linked ticket for what is still missing, which will follow in subsequent PRs.

The old slate code has been copied over to a new file where having both implementations side-by-side wasn't practical. These will be removed once the feature flag is taken off.

To share code with the editing composer, I've moved the basic editing support into `BasicMessageEditor`. Then, there's `EditMessageComposer` using it for editing, and `SendMessageComposer` using it for the main composer. The later is basically the ported version of `MessageComposerInput` to the CIDER editor.

The main composer also needed support for autocompleting commands, which is added through by subclassing the part creator for editing messages.

The CIDER editor also has some documentation at `docs/ciderEditor.md`, which might help the reviewer.

### CODE REVIEW

File: src/components/views/rooms/SendMessageComposer.js [hunk 1]
```diff
@@ -0,0 +1,319 @@
+/*
+Copyright 2019 New Vector Ltd
```
Comment (jryans) [inline src/components/views/rooms/SendMessageComposer.js]: Is this file's content all new? Maybe C.I.C. only if so.

File: src/components/views/rooms/SlateMessageComposer.js [hunk 2]
```diff
@@ -0,0 +1,489 @@
+/*
```
Comment (jryans) [inline src/components/views/rooms/SlateMessageComposer.js]: Hmm, it's unfortunate this isn't detected as a file move... Any guess why it wasn't? If it's moved and edited, maybe the move can happen in a separate commit?

Comment (bwindels) [reply src/components/views/rooms/SlateMessageComposer.js]: Because I copied it over from `develop` after having heavily modified the original on my branch ... That would have been better, I only thought how to keep the slate editor functioning at the end :man_facepalming: As this code will be removed anyway, maybe it is fine?

File: src/components/views/rooms/SendMessageComposer.js (hunk 1 above)
Comment (bwindels) [reply src/components/views/rooms/SendMessageComposer.js]: There is code that was copied over from `MessageComposerInput.js`, so both is probably fine.

File: src/components/views/rooms/SlateMessageComposer.js (hunk 2 above)
Comment (jryans) [reply src/components/views/rooms/SlateMessageComposer.js]: Hmm, depends how you feel about surgery with `git rebase -i` and such... I do find it nice to have things recorded as moves where possible... I am happy to spend a few minutes helping attempt it, as I've done this sort of thing quite often. (But yes, you are correct that the file should be short lived if everything goes according to plan... Sometimes it doesn't though... 😅)
